pkg/loop/reportingplugins/grpc.go

```diff
@@ -73,7 +81,14 @@ func (g *GRPCService[T]) GRPCServer(broker *plugin.GRPCBroker, server *grpc.Serv
 		tc := internal.NewTelemetryClient(ts)
 		return g.PluginServer.NewReportingPluginFactory(ctx, cfg, provider, pr, tc, el)
 	}
-	return internal.RegisterReportingPluginServiceServer(server, broker, g.BrokerConfig, serverAdapter(adapter))
+
+	validateConfigFn := func(ctx context.Context) (types.ValidationService, error) {
+		return g.PluginServer.NewValidationService(ctx)
+	}
+	return internal.RegisterReportingPluginServiceServer(server, broker, g.BrokerConfig, serverAdapter{
```

Comment:
@george-dorin I don't think we should extend the serverAdapter for this; can we instead just embed the PluginServer?

All the adapter should do is translate between the providerConn based NRP function and the provider struct based one. All other methods should be passed through as it.

Reply:
@cedric-cordenier not sure on what you mean when saying embedding the PluginServer, the `GRPCService` is shared between NRP and `NewValidationService`